File: Scripts/Destructable Object.js
#pragma strict

var BrokenObject : Transform;
var Effect : Transform; 
var EffectII :Transform;
var SoundEffect :Transform; 




function OnCollisionEnter (theCollision : Collision){

    if(theCollision.gameObject.tag == "Player")
        BreakDeath();
}

    function BreakDeath(){
        Debug.Log("BreakDeathRead");
        Instantiate(BrokenObject, transform.position,BrokenObject.transform.rotation);
        Instantiate(Effect,transform.position,BrokenObject.transform.rotation);
        Instantiate(EffectII,transform.position,BrokenObject.transform.rotation);
        Destroy(gameObject);
        Instantiate(SoundEffect,transform.position,BrokenObject.transform.rotation);

    }

File: Scripts/DestructableObject.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class DestructableObject : MonoBehaviour {
    public GameObject BrokenObject;
    public GameObject Effect;
    public GameObject Effect2;
    public AudioSource BreakSound;
    // Use this for initialization
    void Start () {
		
	}
    public void OnTriggerEnter(Collider other)
    {
        if (other.gameObject.tag == "Player")
        {
            Instantiate(BrokenObject, transform.position, transform.rotation);
            Instantiate(Effect, transform.position, transform.rotation);
            Instantiate(BreakSound, transform.position, transform.rotation);
            Instantiate(Effect2, transform.position, transform.rotation);
            Destroy(this.gameObject);
        }
    }
    // Update is called once per frame
    void Update () {
		
	}
}
